src/games/supported/VideoPinball.cpp

```diff
@@ -125,3 +129,36 @@ void VideoPinballSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect VideoPinballSettings::getAvailableModes(){
+    ModeVect modes;
+    modes.push_back(0);
+    modes.push_back(2);
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void VideoPinballSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m == 0 || m == 2){
+        m_mode = m;
+        //Read the mode we are currently in
+        unsigned char mode = readRam(&system, 0xC1);
```

Comment:
0xA1 seems to be the correct byte here. 0xC1 is the number of select presses since the last reset. In particular setMode(0) will keep the mode the same (even if it is mode 2). Can you check?